File: clojure/main4.clj
(ns main4)
(defn dangerous [v1 v2] (quot v1 v2))
(defn foo [array counter]
  (if (zero? counter)
    (dangerous (nth array 0) counter)
    (foo array (dec counter))))
(defn -main []
  (let [array (vec (repeat 1000 0))]
    (println (format "The result is %d" (foo array 6)))))
(-main)

File: kotlin/Main4.kt
object Main4  {
    fun dangerous(v1:Int, v2:Int): Int = v1 / v2
    fun foo(array:IntArray, counter:Int): Int =
        if(counter==0) dangerous(array[0], counter) else foo(array, counter-1)
    @JvmStatic fun main(args:Array<String>){
        val array = IntArray(1000){0}
        println("The result is ${foo(array, 6)}")
    }
}

